tuxemon/states/combat/combat.py

```diff
@@ -1149,6 +1152,21 @@ def trigger_player_run(self, player: NPC) -> None:
         del self.monsters_in_play[player]
         self.players.remove(player)
 
+    def evolve(self, player: NPC, monster: Monster) -> None:
+        for evolution in monster.evolutions:
+            # check the path field.
+            if not evolution["path"]:
+                # check the level field.
+                if evolution["at_level"] <= monster.level:
+                    logger.info(
+                        "{} evolved into {}!".format(
+                            monster.name, evolution["monster_slug"]
+                        )
+                    )
+                    monster_evolve = str(evolution["monster_slug"])
+                    if player is self.players[0]:
```

Comment:
move this constraint into the calling code

Reply:
I rewrote that part.